src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -2009,9 +2043,135 @@ public String listOperations(String clusterName, OpType operationType, String ho
   @Override
   public void purgeNodeOperations() {}
 
+  @Override
+  public boolean lockRunningRepairsForNodes(
+      UUID repairId,
+      UUID segmentId,
+      Set<String> replicas) {
+    List<ResultSetFuture> futures = Lists.newArrayList();
+    for (String replica:replicas) {
+      // Maybe initialize the row for each node for that repair
+      futures.add(session.executeAsync(initRunningRepairsPrepStmt.bind(repairId, replica)));
+    }
+    try {
+      Futures.allAsList(futures).get();
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.warn("Woah, that didn't feel right...", e);
+    }
+
+    // Attempt to lock all the nodes involved in the segment
+    BatchStatement batch = new BatchStatement();
+    for (String replica:replicas) {
+      batch.add(
+          setRunningRepairsPrepStmt.bind(
+              LEAD_DURATION,
+              AppContext.REAPER_INSTANCE_ADDRESS,
+              reaperInstanceId,
+              segmentId,
+              repairId,
+              replica,
+              null));
+    }
+
+    ResultSet results = session.execute(batch);
+    if (!results.wasApplied()) {
+      logFailedLead(results, repairId);
+    }
+
+    return results.wasApplied();
+  }
+
+  @Override
+  public boolean renewRunningRepairsForNodes(
+      UUID repairId,
+      UUID segmentId,
+      Set<String> replicas) {
+    // Attempt to renew lock on all the nodes involved in the segment
+    BatchStatement batch = new BatchStatement();
+    for (String replica:replicas) {
+      batch.add(
+          setRunningRepairsPrepStmt.bind(
+              LEAD_DURATION,
+              AppContext.REAPER_INSTANCE_ADDRESS,
+              reaperInstanceId,
+              segmentId,
+              repairId,
+              replica,
+              reaperInstanceId));
+    }
+
+    ResultSet results = session.execute(batch);
+    if (!results.wasApplied()) {
+      logFailedLead(results, repairId);
+    }
+
+    return results.wasApplied();
+  }
+
+  private void logFailedLead(ResultSet results, UUID repairId) {
+    LOG.debug("Failed taking/renewing lock for repair {} because segments are already running for some nodes.",
+        repairId);
+    for (Row row:results) {
+      LOG.debug("node {} is locked by {}/{} for segment {}",
+          row.getColumnDefinitions().contains("node")
+            ? row.getString("node")
+            : "unknown",
+          row.getColumnDefinitions().contains("reaper_instance_host")
+            ? row.getString("reaper_instance_host")
+            : "unknown",
+          row.getColumnDefinitions().contains("reaper_instance_id")
+            ? row.getUUID("reaper_instance_id")
+            : "unknown",
+          row.getColumnDefinitions().contains("segment_id")
+            ? row.getUUID("segment_id")
+            : "unknown"
+      );
```

Comment:
nit: i'd find this more readable as 
```suggestion
      LOG.debug("node {} is locked by {}/{} for segment {}",
          row.getColumnDefinitions().contains("node") ? row.getString("node") : "unknown",
          row.getColumnDefinitions().contains("reaper_instance_host") ? row.getString("reaper_instance_host") : "unknown",
          row.getColumnDefinitions().contains("reaper_instance_id") ? row.getUUID("reaper_instance_id") : "unknown",
          row.getColumnDefinitions().contains("segment_id") ? row.getUUID("segment_id") : "unknown"
      );
```

Reply:
Agreed, but it breaks our code style as some lines would go over 120 characters 🤷